### ISSUE TITLE
Adaptive Response Action Cannot Be Dispatched, Unexpected Token < in JSON at position 0

### ISSUE THREAD

**Issue by tcsalone** [guilhemmarchand/TA-jira-service-desk-simple-addon#81]
Hello, 

Trying to find an easy way to create JIRA tickets from noteable events in ES.  Installed "JIRA Service Desk simple addon" v 1.0.26.  When we try to create a ticket, we get an error "Adaptive Response Action Cannot Be Dispatched, Unexpected Token < in JSON at position 0".  

Basic connectivity seems to work, the app is able to read info from our JIRA instance and populate some dropdown menus.  JIRA is hosted in the cloud, we're also on Splunk Cloud 8.1  

Turned on debug, dont see any content in the error reporting seraching within the app.  Any ideas on where to troubleshoot?

![jira_desk_error](https://user-images.githubusercontent.com/7087767/116949033-f181ec80-ac35-11eb-85c1-61e15f104c65.PNG)

**Comment by guilhemmarchand** [comment]
Hi @tcsalone 

I will check it out, and will let you know ASAP, the adaptive response action per incident review was working fine earlier, so there might have been a regression I haven't noticed.

However, the best approaches to create JIRA ticket from notable events in a programmatic and automated fashion are the following:

1. Create a global alert looking at notable events

Create an alert which enables the JIRA alaert action and looks at the notable events out of ES, say "Notables to JIRA", in this alert that you create in another application name space (like the search app), you takes all notables, then you can for instance filters on notables eventually that match a specific criterias, say:

```
`notables` | search source=*<my source filter>*
```

A common approach is to maintain in a lookup a list of correlation searches for which the JIRA ticket creation should be enabled, then filter against it, say

```
`notables` | lookup <my lookup> search_name OUTPUT enable_jira | where enable_jira="true"
```

(this is an example for the explanation purposes)

Then you make sure the alert is configured to trigger for each results, you configure properly the alert action, you can define in the SPL or the lookup additional information you would want to submit while creating the ticket recycling the row results as variables (ex: $result.myfield$) and that's it ;-)

This is approach is the better design and PS recommended.

2. Enable the alert action per correlation search

You can as well enable the alert action per correlation search, use the form to set the JIRA options as needed, the trap is to make sure to enable "Trigger for each result" which as to be  made out of ES content management UI as it is a different option (you need to go via Settings / Searches, reports and alerts)

So, to summarise, the issues for which you opened is the creation of notable as per a manual action in IR, this is fine and should be working so I will check, but it's not the common way to create JIRAs from notables.

Guilhem

**Comment by guilhemmarchand** [comment]
@tcsalone 

I have tried unsuccessfully to replicate a failure while call the Adaptive Response from Incident Review (not in Splunk Cloud thus), and things are working as they should:

<img width="1779" alt="image" src="https://user-images.githubusercontent.com/7503059/117566397-57df9280-b0ae-11eb-92d7-b38326a17967.png">

<img width="695" alt="image" src="https://user-images.githubusercontent.com/7503059/117566411-6af26280-b0ae-11eb-8315-d45289bfe8cc.png">

<img width="699" alt="image" src="https://user-images.githubusercontent.com/7503059/117566433-89585e00-b0ae-11eb-9d0d-ccf77d0ee08e.png">

<img width="893" alt="image" src="https://user-images.githubusercontent.com/7503059/117566447-970de380-b0ae-11eb-8dec-28bfe9d395c4.png">

<img width="1777" alt="image" src="https://user-images.githubusercontent.com/7503059/117566473-aa20b380-b0ae-11eb-8e7e-048219f22f38.png">

<img width="1781" alt="image" src="https://user-images.githubusercontent.com/7503059/117566501-d2101700-b0ae-11eb-874d-24130cf5512d.png">

<img width="1789" alt="image" src="https://user-images.githubusercontent.com/7503059/117566525-f4a23000-b0ae-11eb-9329-53163b253d4a.png">

- Do you have any progress regarding my previous comments?
- Which version of the app is running Splunk Cloud?

It could be eventually a Coud thing if the Cloud automation removes some stuffs it shouldn't, but a bit unlikely.

**Comment by guilhemmarchand** [comment]
Any update @tcsalone ?

**Comment by guilhemmarchand** [comment]
Closing with no response - if still an issue please re-open

**Comment by mrigankchanda** [comment]
Hi @guilhemmarchand, I recently installed this app on my Splunk Cloud instance, and I am in the same situation as OP.

App is setup and able to get project and task data into Jira dashboards on Splunk
However, the scheduled search action and the Notable adaptive response action are not pushing task creation. The error is almost the same
splunk version: 9.3.2411.114
ES version: 7.3.4
<img width="783" height="193" alt="Image" src="https://github.com/user-attachments/assets/d69e50ad-2fb7-4f50-ac91-d8e42312bbe2" />

**Comment by guilhemmarchand** [comment]
Hi @mrigankchanda 

To be clear this is limited to the adaptive response under Enterprise Security isn't it? (as a manual action)
Can you confirm that you can create ticket using the alert action in a standard alert?

Guilhem

**Comment by mrigankchanda** [comment]
Thanks for the quick response.

I tried to run this using a scheduled search alert, with an alert action to create a task
Below is the failure message

After removing the priority, it's working fine from the alert action from a standard alert

I was hoping for it to work via manual action as well, for someone to run it from the notable's adaptive response, selectively

**Comment by mrigankchanda** [comment]
Hey @guilhemmarchand , any luck finding a resolution to use it via adaptive response from the incident review page?

**Comment by guilhemmarchand** [comment]
Hi @mrigankchanda !

I believe this to be fixable indeed, I need to review the basis configuration that should generated by the Splunk UCC framework and which is specific to adaptive responses, I will try to look after this within before the end of week and will let you know 

Guilhem

**Comment by mrigankchanda** [comment]
Thanks @guilhemmarchand

**Comment by mmacielinski** [comment]
> Hi [@guilhemmarchand](https://github.com/guilhemmarchand), I recently installed this app on my Splunk Cloud instance, and I am in the same situation as OP.
> 
> App is setup and able to get project and task data into Jira dashboards on Splunk However, the scheduled search action and the Notable adaptive response action are not pushing task creation. The error is almost the same splunk version: 9.3.2411.114 ES version: 7.3.4 <img alt="Image" width="783" height="193" src="https://private-user-images.githubusercontent.com/65531199/483484364-d69e50ad-2fb7-4f50-ac91-d8e42312bbe2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkxNTg0MjYsIm5iZiI6MTc1OTE1ODEyNiwicGF0aCI6Ii82NTUzMTE5OS80ODM0ODQzNjQtZDY5ZTUwYWQtMmZiNy00ZjUwLWFjOTEtZDhlNDIzMTJiYmUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI5VDE1MDIwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjYzJhZDRlMTJmNGI0MTEwNDA3YjAzZmY5YjcyOWQ1YzY5OWE2ZDMyYTE1MzA3YWUzZjgwYjQ2NDgyZDE0NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.E6AseApppDbVT3kZjM293bFutBkP95xEEN-PkdCO0cc">


Testing this with Splunk Enterprise 9.1.2 and ES 7.3.3 adaptive response and getting the exact same error message.
Only filling in the required fields for initial testing.  Version 2.1.1 of the atlassian app.
